libr/core/cmd_api.c

```diff
@@ -355,6 +355,157 @@ R_API RCmdStatus r_cmd_call_parsed_args(RCmd *cmd, RCmdParsedArgs *args) {
 	return res;
 }
 
+static size_t strlen0(const char *s) {
+	return s? strlen (s): 0;
+}
+
+static void fill_usage_strbuf(RStrBuf *sb, RCmdDesc *cd) {
+	r_strbuf_append (sb, "Usage: ");
+	if (cd->help.usage) {
+		r_strbuf_appendf (sb, "%s", cd->help.usage);
+	} else {
+		r_strbuf_appendf (sb, "%s %s", cd->name, cd->help.args_str);
+	}
+	if (cd->help.group_summary) {
+		r_strbuf_appendf (sb, "   # %s\n", cd->help.group_summary);
+	} else {
+		r_strbuf_appendf (sb, "   # %s\n", cd->help.summary);
+	}
+}
+
+static size_t update_max_len(RCmdDesc *cd, size_t max_len) {
+	size_t name_len = strlen (cd->name);
+	size_t args_len = strlen0 (cd->help.args_str);
+	if (name_len + args_len > max_len) {
+		return name_len + args_len;
+	}
+	return max_len;
+}
+
+static void print_child_help(RStrBuf *sb, RCmdDesc *cd, size_t max_len) {
+	size_t str_len = strlen (cd->name) + strlen0 (cd->help.args_str);
+	size_t padding = str_len < max_len ? max_len - str_len : 0;
+	const char *cd_args_str = cd->help.args_str ? cd->help.args_str : "";
+	const char *cd_summary = cd->help.summary ? cd->help.summary : "";
+	r_strbuf_appendf (sb, "| %s %s %*s# %s\n", cd->name, cd_args_str, padding, "", cd_summary);
```

Comment:
If `args_str` was printed without implicit space it could also be used for specifying "[a] <arg>" part from "cmd[a] <arg>" when it makes sense.

Reply:
I am also not sure if it's necessary to distinguish "usage" from "{cmd}{args_str}".

Reply:
Let me try this.

Reply:
I am not sure this is possible as currently the "usage" part may be different from the command name sometimes. Also, what do you mean with `If args_str was printed without implicit space it could also be used for specifying "[a] " part from "cmd[a] " when it makes sense.`?